fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluid/v1/FabricFlowableFluid.java

```diff
@@ -0,0 +1,87 @@
+import net.minecraft.client.render.FogShape;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.Entity;
+import net.minecraft.fluid.FlowableFluid;
+import net.minecraft.sound.SoundEvent;
+
+/**
+ * Extends the {@link FlowableFluid} class by adding more features.
+ */
+public abstract class FabricFlowableFluid extends FlowableFluid {
+	/**
+	 * Gets the color of the fluid fog.
+	 *
+	 * @param player    Player submerged by the fluid.
+	 * @param tickDelta Time passed from the last tick.
+	 * @param world     The current client-side world.
+	 * @return <p>An int value indicating the color of the fluid fog.</p>
+	 * <p>If the color is -1 no fog will be rendered.</p>
+	 * <p>(It could be a hexadecimal value like 0xFFFFFF).</p>
+	 */
+	public abstract int getFabricFogColor(Entity player, float tickDelta, ClientWorld world);
+
+	/**
+	 * <p>Gets the distance in blocks, from the player camera position, in which the fog starts rendering.</p>
+	 * <p>This could be negative, in this case the fog starts partially opaque.</p>
+	 *
+	 * @param player       Player submerged by the fluid.
+	 * @param fogType      Type of the fog (SKY or TERRAIN).
+	 * @param viewDistance View distance of the player in blocks.
+	 * @param thickFog     Specifies if the fog is thick.
+	 * @return A float indicating the start distance of the fluid fog from the player camera, in blocks.
+	 */
+	public abstract float getFabricFogStart(Entity player, BackgroundRenderer.FogType fogType, float viewDistance, boolean thickFog);
+
+	/**
+	 * <p>Gets the distance in blocks, from the player camera position, in which the fog is totally opaque.</p>
+	 * <p>If the end distance is less than the start distance, it will be ignored.</p>
+	 *
+	 * @param player       Player submerged by the fluid.
+	 * @param fogType      Type of the fog (SKY or TERRAIN).
+	 * @param viewDistance View distance of the player in blocks.
+	 * @param thickFog     Specifies if the fog is thick.
+	 * @return A float indicating the end distance of the fluid fog from the player camera, in blocks.
+	 */
+	public abstract float getFabricFogEnd(Entity player, BackgroundRenderer.FogType fogType, float viewDistance, boolean thickFog);
+
+	/**
+	 * Gets the shape of the fluid fog.
+	 *
+	 * @param player       Player submerged by the fluid.
+	 * @param fogType      Type of the fog (SKY or TERRAIN).
+	 * @param viewDistance View distance of the player in blocks.
+	 * @param thickFog     Specifies if the fog is thick.
+	 * @return An enum value indicating the shape of the fluid fog: CYLINDER or SPHERE.
+	 */
+	public abstract FogShape getFabricFogShape(Entity player, BackgroundRenderer.FogType fogType, float viewDistance, boolean thickFog);
+
+	/**
+	 * Gets the sound played when the bucket containing this fluid is emptied.
+	 *
+	 * @return SoundEvent played when the bucket containing this fluid is emptied.
+	 */
+	public Optional<SoundEvent> getFabricBucketEmptySound() {
```

Comment:
We could also put this directly to `FlowableFluid` with an injected interface like this:

```java
interface FabricFlowableFluid {
  default Optional<SoundEvent> getFabricBucketEmptySound() { return Optional.empty(); }
}
```

Reply:
Check [Loom's 0.11 changelog](https://github.com/FabricMC/fabric-loom/releases/tag/0.11) for how to do it.

Reply:
Was just a bit tricky but works better than expexted.
I think that the wiki must be updated, because there is no example to use interface injections correctly.